client/src/js/users/components/__tests__/Create.test.js

```diff
@@ -1,71 +1,182 @@
+        });
+    });
+});
+
+describe("mapStateToProps", () => {
+    it("should call mapStateToProps", () => {
+        const state = {
+            errors: {
+                CREATE_USER_ERROR: {
+                    message: "foo"
+                }
+            },
+            router: { location: "foo" },
+            users: {
+                createPending: true
+            },
+            settings: {
+                data: {
+                    minimum_password_length: 1
+                }
+            }
         };
-        const store = mockStore(initialState);
-        let spy;
 
-        afterEach(() => {
-            spy.restore();
+        const result = mapStateToProps(state);
+        expect(result).toEqual({
+            show: false,
+            pending: true,
+            minimumPasswordLength: 1,
+            error: "foo"
+        });
+    });
+});
+
+describe("mapDispatchToProps", () => {
+    let dispatch;
+    let result;
+
+    beforeEach(() => {
+        dispatch = jest.fn();
+        result = mapDispatchToProps(dispatch);
+    });
+
+    it("should return onCreate() in props", () => {
+        const data = "foo";
+
+        result.onCreate(data);
+        expect(dispatch).toHaveBeenCalledWith({
+            0: "f",
+            1: "o",
+            2: "o",
+            type: "CREATE_USER_REQUESTED"
         });
+    });
+    it("should return onHide() in props", () => {
+        push.mockReturnValue({
+            type: "PUSH"
+        });
+
+        result.onHide({ ...window.location, state: { createUser: false } });
+
+        expect(dispatch).toHaveBeenCalledWith({ type: "PUSH" });
+    });
 
-        it("Submitting form dispatches createUser() action", () => {
-            spy = sinon.spy(actions, "createUser");
-            expect(spy.called).toBe(false);
-
-            wrapper = mount(<CreateUserContainer store={store} />);
-            wrapper
-                .find({ label: "Username" })
-                .at(0)
-                .prop("onChange")({ target: { name: "userId", value: "test-user" } });
-            wrapper
-                .find({ label: "Password" })
-                .at(0)
-                .prop("onChange")({
-                target: { name: "password", value: "test-password" }
-            });
-            wrapper
-                .find({ label: "Confirm" })
-                .at(0)
-                .prop("onChange")({
-                target: { name: "confirm", value: "test-password" }
-            });
-
-            wrapper.find("form").prop("onSubmit")({ preventDefault: jest.fn() });
-
-            expect(
-                spy.calledWith({
-                    userId: "test-user",
-                    password: "test-password",
-                    confirm: "test-password",
-                    forceReset: false
-                })
-            ).toBe(true);
+    it("should return onClearError() in props", () => {
+        const error = true;
+        const onClearError = jest.fn();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['onClearError' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/virtool/virtool/pullRequest?prid=4626071)